app/controllers/concerns/api/v3/sync_to_user.rb

```diff
@@ -61,11 +64,27 @@ def current_facility_processed_since
     end
 
     def force_resync?
-      process_token[:resync_token] != resync_token
+      resync_token_modified? || sync_region_modified?
     end
 
-    def resync_token
-      request.headers["HTTP_X_RESYNC_TOKEN"]
+    def resync_token_modified?
+      process_token[:resync_token] != current_resync_token
+    end
+
+    def sync_region_modified?
+      process_token[:sync_region_id] != current_sync_region_id
+    end
+
+    def sync_region
+      if block_level_sync?
+        current_block
+      else
+        current_facility_group
+      end
+    end
+
+    def block_level_sync?
+      current_user.feature_enabled?(:region_level_sync) && process_token[:sync_region_id].present?
```

Comment:
Check if `requested_sync_region_id` is present instead of `process_token[:sync_region_id].present?`.